### DISCUSSION TITLE
DMS fonts not loading

### DISCUSSION THREAD

**Discussion by AndrewMC6** [mnit-rtmc/iris#160] [Q&A]
Hello, We've recently been working on getting IRIS working in our environment, we've been able to install the latest RPM Iris-5.50.1-1.noarch.rpm and are starting to populate the map with our infrastructure. We are running into an issue with DMS fonts not loading after performing the "Query Settings" as described in the documentation. the WYSIWYG view is greyed out, and the config window says Font (1) is not in the database. We can see fonts being downloaded into folders, but there seems to be something missing from the documentation about how to load these fonts. We found the tfon_import.py missing from the latest release, found it in a previous release and ran it, with still no change in this behavior.

Happy to provide some more information, thanks!

**Comment by DougLau** [comment]
As of v5.46, any required fonts must be imported into the database.  See https://mnit-rtmc.github.io/iris/fonts.html

You can use any of the fonts supplied with IRIS, or any fonts built into a sign.  To use a sign font:
- After **Query Settings** is completed successfully, there should be a directory on the server at  `/var/lib/iris/web/tfon/{sign_name}`.  It will contain fonts from the sign in `.tfon` format
- The font can be imported into the database using the instructions at https://mnit-rtmc.github.io/iris/fonts.html#importing
- Note: each font must have a unique number in IRIS

**Comment by dmcrews4202** [comment]
Here is what I saw.

After Query Settings, the fonts were being downloaded into a folder: /var/lib/iris/fonts/{sign_name}

I moved them to /var/lib/iris/web/tfon/{sign_name}

I imported them into the database, and verified the records were there

Trying to use the message editor, create button there was no change. WYSIWYG is greyed out, and config showed "Default font (1) not in database.  Error loading configuration for sign 48 South - MCRR"

The one thing I do see is that the folders that are created have spaces in the name. For example /var/lib/iris/fonts/'40 South - MCRR'

Could that be the issue?

**Reply by DougLau** [reply]
You may need to set the default font for that sign's configuration: https://mnit-rtmc.github.io/iris/sign_configuration.html

**Reply by dmcrews4202** [reply]
Thanks for your help. I believe we are seeing fonts now. What I was missing was bouncing the IRIS server after making the font changes. It hadn't picked up the changes yet.

**Reply by AndrewMC6** [reply]
I confirm we are seeing fonts and the WYSIWYG editor, discovered now there is an issue with Message Patterns and the message selector. Is there a way to allow messages to show in the message selector list without compose # tags, or when created from the message selector, they automatically get a certain compose # ?

Is there no way for an operator to be able to do a "quick message" to a sign? IE a crash report comes in and they'd immediately be able to open a WYSIWYG and key in/send a message right out?

**Reply by DougLau** [reply]
The free-form message feature has changed a couple of times.  In version 5.50, it requires the user's role to have **Manage** permission for signs.  But there is not a simple UI to enable that configuration.

In the current version (5.61) permissions have been simplified, with a new UI along with many other enhancements.  I will build an RPM in case you want to upgrade.

**Comment by AndrewMC6** [comment]
If you could that would be great, the software we’re looking at migrating from has free-form messages, which our TMC staff use frequently so it would be an important feature for us. An RPM would be helpful. Thanks!


Sent from my iPhone

On Oct 31, 2024, at 09:08, Douglas Lau ***@***.***> wrote:

﻿
The free-form message feature has changed a couple of times. In version 5. 50, it requires the user's role to have Manage permission for signs. But there is not a simple UI to enable that configuration. In the current version (5. 61) permissions


The free-form message feature has changed a couple of times. In version 5.50, it requires the user's role to have Manage permission for signs. But there is not a simple UI to enable that configuration.

In the current version (5.61) permissions have been simplified, with a new UI along with many other enhancements. I will build an RPM in case you want to upgrade.

**Reply by DougLau** [reply]
http://iris.dot.state.mn.us/rpms/iris-5.61.1-1.noarch.rpm

**Reply by AndrewMC6** [reply]
Thanks Doug, we've applied this patch however I'm not seeing any UI changes, and don't see freeform/quick messages, we're currently testing with administrator accounts that have all configure access. I did try changing DMS to manage, and didn't see any change in options. Am I missing something?

thanks!

**Reply by DougLau** [reply]
My guess is you need to add a [hashtag](https://mnit-rtmc.github.io/iris/hashtags.html) to the DMS `notes` field that matches a message pattern's compose hashtag.  By default, there are patterns with `#OneLine`, `#TwoLine`, `#ThreeLine` and `#FourLine` that could be used.

**Comment by AndrewMC6** [comment]
Is there no way for a message to automatically be given a tag? Right now it seems like you’d have to create a message in the editor, save and close, then go into the message library and set the tag manually.

There is no way other workflow for creating a message? I’d be curious if there was a way for an operator to right click or double click a VMS and a message creation dialog pops up and they can immediately push something out for incidents, the message library looks useful for preformed messages about winter weather etc.


Sent from my iPhone

On Nov 5, 2024, at 09:02, Douglas Lau ***@***.***> wrote:

﻿
My guess is you need to add a hashtag to the DMS notes field that matches a message pattern's compose hashtag. By default, there are patterns with #OneLine, #TwoLine, #ThreeLine and #FourLine that could be used. — Reply to this email directly,


My guess is you need to add a hashtag<https://urldefense.com/v3/__https://mnit-rtmc.github.io/iris/hashtags.html__;!!OOzr21-N2k4!ZTYZ2tuMsfQWyVmRxXJv1zT-tjR_hHbOXEiifxe3nrszKLN0hGEhSNeCNI4RbJHKgk3-eAjGoRJd6qCD1GvOFSwjQCYkwcY$> to the DMS notes field that matches a message pattern's compose hashtag. By default, there are patterns with #OneLine, #TwoLine, #ThreeLine and #FourLine that could be used.

**Reply by DougLau** [reply]
Typically, a message pattern would have [fillable text rectangles](https://mnit-rtmc.github.io/iris/message_patterns.html#fillable-text-rectangles).  This allows the operator to select "message lines" from selectors to build messages based on that pattern.  Additionally, ff the operator has free-form (Manage) permissions, they can type/edit the message directly into the selectors.
